Enhancement: Warn about sync dependencies without an explicit `sync_to_thread` value (#1648)
litestar-org/litestar: docs/usage/dependency-injection.rst

   from litestar.di import Provide

>    async def bool_fn() -> bool:

Reviewer: This file is a good place to demo the different patterns and explain them

:meth:`object.__call__` method, or classes. These are in turn wrapped inside an instance of the
:class:`Provide <.di.Provide>` class.

> .. admonition:: Sync vs. Async dependencies
>     :class: important
> 
>     Litestar supports both synchronous and asynchronous dependencies. To ensure
>     To ensure synchronous dependencies don't block the main thread and therefore the
>     entire application (for example when they perform blocking I/O), the parameter
>     ``sync_to_thread`` will cause them to be run in a thread pool. If a synchronous
>     function is non-blocking, it should be considered to make it an async function
>     instead.

Reviewer 2: ```suggestion
.. admonition:: Sync vs. Async dependencies
    :class: important

    Litestar supports both **synchronous** and **asynchronous** dependencies.
    To ensure **synchronous** dependencies don't block the main thread
    - *and therefore the entire application, for example when they perform blocking I/O* -
    the parameter ``sync_to_thread`` will ensure they are run in a thread pool.
    If a **synchronous** function is non-blocking, you should consider making it an async function instead.
```

.. tip::

>     Litestar will warn about the usage of synchronous dependency functions which don't
>     have ``sync_to_thread`` set.

Reviewer 2: ```suggestion
    Litestar will warn about the usage of synchronous dependency functions which do not
    have ``sync_to_thread`` set.
```
